### PR TITLE
chore: updated lsp-types & some minimum fixes

### PR DESCRIPTION
**Pull request by Chronostasys**

### CODE REVIEW

File: src/ast/ctx/lsp.rs [hunk 1]
```diff
@@ -104,11 +106,24 @@ impl Ctx<'_> {
         if self.need_highlight.borrow().ne(&0) || self.in_macro {
             return;
         }
+        let part1 = InlayHintLabelPart {
```
Comment (xieyuschen) [inline src/ast/ctx/lsp.rs:109]: consider to rename the `part1` and `part2` to describe what are they.

File: src/ast/node/function.rs [hunk 2]
```diff
@@ -185,7 +185,12 @@ impl FuncCallNode {
     fn build_hint(&mut self, ctx: &mut Ctx, fnvalue: &FNValue, skip: u32) {
         for (i, para) in self.paralist.iter_mut().enumerate() {
             let pararange = para.range();
-            ctx.push_param_hint(pararange, fnvalue.param_names[i + skip as usize].clone());
+            let mut b = FmtBuilder::default();
+            para.format(&mut b);
+            let para_s = b.generate();
```
Comment (xieyuschen) [inline src/ast/node/function.rs:190]: can consider to rename it as `param_name` or something else. I'm not sure what's the `_s` means

Comment (Chronostasys) [reply src/ast/node/function.rs:190]: it means str

**Review by xieyuschen — DISMISSED** [review]
LGTM:)

**Pull request merged**